### PR TITLE
create edit password functionality

### PR DESCRIPTION
**Pull request by MontherIsmail**
Relates #50
what route should do
-edit password for buyer and seller

-method: POST

-Route: /api/v1/profile/

Protected: Yes

Required middleWares: asyncMiddleware, checkAuth
Response
Successs  : status : 201, {message : password edited } .
Failure : status : 401 ,  {message : validation fails} . // validation fails
Failure : status : 400 ,  {message : enter new password} . // same old password
Failure : status 500 , {message : internal server error } //server error
Request body: if exists
{
newPassword

}

        : is exists
{
  oldPassword: string().required()
  newPassword: string().required()
}

Tests:
 all our status code
 201 => insert in data base 
401 => validation fails
400 =>bad request
503  =>service unavailable
500 => server error

### CODE REVIEW

**Review by Fady-Alwazir — APPROVED** [review]

**Review by mayar-kabaja — APPROVED** [review]

**Review by WessalJawad95 — APPROVED** [review]

File: server/src/controllers/profile/editPassword.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { editPasswordDB, getUserById } from '../../database/queries';
+import { comparePassword, editPasswordSchema, hashPassword } from '../../utils';
+
+const editPassword = async (req, res) => {
+  const { oldPassword, newPassword } = req.body;
+  const id = 1;
```
Comment (Nizar7zak) [inline server/src/controllers/profile/editPassword.ts:6]: I suppose you forgot get userId from request, not as static data !

File: server/src/controllers/profile/editPassword.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  await editPasswordSchema.validateAsync(req.body);
+  const { rows } = await getUserById(id);
+  const { password } = rows[0];
+  const isMatch = await comparePassword(oldPassword, password);
+  if (!isMatch) {
+    return res.status(400).json({ message: 'wrong password !' });
+  }
+  const hashedNewPassword = await hashPassword(newPassword);
```
Comment (Nizar7zak) [inline server/src/controllers/profile/editPassword.ts:14]: here, you should check if the old password equal new password, then you will send a response to tell the client side your new password matching your old password.

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

**Review by Fady-Alwazir — APPROVED** [review]

**Review by Nizar7zak — APPROVED** [review]
Good job Monther.

File: server/src/controllers/profile/editPassword.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { editPasswordDB, getUserById } from '../../database/queries';
+import { comparePassword, editPasswordSchema, hashPassword } from '../../utils';
+
+const editPassword = async (req, res) => {
+  const { oldPassword, newPassword } = req.body;
+  const { id } = req;
```
Comment (Nizar7zak) [inline server/src/controllers/profile/editPassword.ts:6]: from checkAuth middleware we take a userId not id here.

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

**Pull request merged**